packages/integration-sdk-runtime/src/storage/FileSystemGraphObjectStore/FileSystemGraphObjectStore.ts

```diff
@@ -91,7 +91,7 @@ function integrationStepsToGraphObjectIndexMetadataMap(
       relationships: new Map(),
     };
 
-    for (const entityMetadata of step.entities) {
+    for (const entityMetadata of step.entities || []) {
```

Comment:
Isn't this typed in such a way that we always require at least an empty array?

Reply:
It is. We shouldn't

Reply:
Pushed an additional commit for this

Reply:
Why shouldn't we?
It's easy to search for
```
entities: []
```
But very difficult to search for it's absence.

Reply:
Once you're inside an executionHandler it's easy to just see what doesn't produce entities; it's kinda the same as you do but the opposite, search for entities and if not found then no entities, this just came up as I was trying to make a dummy step and having to add entities/relationships seems like an extra step that adds no value.

With that being said, moving to draft

Reply:
btw -- same logic you commented would apply for mappedRelationships